src/app-layer-parser.c

```diff
@@ -880,10 +879,35 @@ extern int g_detect_disabled;
 extern bool g_file_logger_enabled;
 extern bool g_filedata_logger_enabled;
 
+// maximum size of allocated array, and thus maximum number
+// of transactions that can be cleant in one run
+#define TX_LIST_MAX 1024
+
+static void AppLayerPrepareCleanupTxList(AppLayerCleanupTxList *txl, uint64_t nbtx)
```

Comment:
https://github.com/OISF/suricata/pull/11555/files#diff-905c3f1e9825e31d7e1b362953f18f0df60680c15a5a00fcd437f6d4f30dde89R923 had a `free_all` logic for when certain conditions weren't met, but I don't see that anymore. Shouldn't that be brought back?

Also, do we really limit to 1024. It seems we check for > 1024 and then resize to the larger number?